src/UhdmAst.cpp

```diff
@@ -1103,6 +911,206 @@ AstNode* process_operation(vpiHandle obj_h, UhdmShared& shared) {
     return nullptr;
 }
 
+AstNode* process_assignment(vpiHandle obj_h, UhdmShared& shared) {
+    AstNode* lvaluep = nullptr;
+    AstNode* rvaluep = nullptr;
+    const unsigned int objectType = vpi_get(vpiType, obj_h);
+    const unsigned int operationType = vpi_get(vpiOpType, obj_h);
+
+    // Right
+    visit_one_to_one({vpiRhs}, obj_h, shared, [&](AstNode* childp) { rvaluep = childp; });
+
+    // Left
+    visit_one_to_one({vpiLhs}, obj_h, shared, [&](AstNode* childp) { lvaluep = childp; });
+
+    const unsigned int arithmeticAssignments[] = {vpiBitAndOp, vpiBitOrOp, vpiBitXorOp,
+                                                  vpiBitXnorOp, vpiLogAndOp, vpiLogOrOp,
+                                                  vpiSubOp, vpiAddOp, vpiMultOp, vpiDivOp,
+                                                  vpiModOp, vpiArithLShiftOp, vpiLShiftOp,
+                                                  vpiRShiftOp, vpiArithRShiftOp};
```

Comment:
Variable name - those are not assignment types, but allowed operations. Maybe something along the lines of `arithmeticAssignmentOpTypes` would be better?

Reply:
Yes, it would be better.
 First I named it arithmeticAndLogicalAssignmentOperations, and then tried to make it shorter.